mods/Fifty.ServerTeamBalance/mod/scripts/vscripts/fstb_command_callbacks.nut

```diff
@@ -12,33 +12,35 @@ void function FSTM_CommandCallback_Switch( entity player, array< string > args )
 		return
 	}
 
-	entity target = player
-
+	entity target
 	if( FSA_IsAdmin( player ) && args.len() != 0 ) {
-		bool foundTarget = false
 		foreach( entity p in GetPlayerArray() ) {
 			if( p.GetPlayerName().tolower().find( args[0].tolower() ) != -1 ) {
+				if( target != null ){
+					FSU_PrivateChatMessage(player, "%EMore than one matching player! %TWrite a bit more of their name.")
+					return
+				}
 				target = p
-				foundTarget = true
-				break
 			}
 		}
-
-		if( !foundTarget ) {
+		if(target == null){
 			FSU_PrivateChatMessage( player, "%H\"" + args[0] + "\"%E couldn't be found!" )
 			return
 		}
+		else
+			FSU_PrivateChatMessage( player, "%S The team of %H" + target.GetPlayerName() + "%S has been switched!" )
 	}
+	else
+		target = player
 
 	if( !IsValid( target ) ) {
-		FSU_PrivateChatMessage( player, "Player entity isn't valid!" )
+		FSU_PrivateChatMessage( player, "%EThat player entity isn't valid!" )
 		return
 	}
 
-	if( IsAlive( player ) )
+	if( IsAlive(player) && player == target )
```

Comment:
`player` is the one calling this, `target` is the one being switched. They may not be the same entity so this is wrong

Reply:
They are the same entity whenever !switch is used normally

They are not the same entity, when an admin is switching someone elses team, in which case killing the target is unnecessary, and could even be disruptive. Hence the check.